crates/wit-component/src/encoding/wit.rs

```diff
@@ -272,11 +272,15 @@ impl InterfaceEncoder<'_> {
         }
     }
 
-    fn encode_instance(&mut self, interface: InterfaceId) -> Result<u32> {
+    fn encode_instance_contents(&mut self, interface: InterfaceId) -> Result<InstanceType> {
         self.push_instance();
         let iface = &self.resolve.interfaces[interface];
         let mut type_order = IndexSet::new();
         for (_, id) in iface.types.iter() {
+            let ty = &self.resolve.types[*id];
+            if let TypeOwner::Interface(iface_id) = ty.owner {
+                self.interface = Some(iface_id);
+            }
```

Comment:
I think this might be best done a little differently than updating `self.interface` here. What I'm worried about is that this updates `self.interface` but doesn't restore it what it previously was. 

Looking throughout this code though `push_instance` and `pop_instance` might be a good place to hook into? I think it should be ok to move most modifications of `self.interface` into those functions where push takes an `InterfaceId` and sets `self.interface` and then pop restores the previous value saved during push.